src/proxy/http_context.rs

```diff
@@ -191,7 +191,11 @@ impl HttpAuthThreescale {
         self.add_http_request_header("x-3scale-upstream-url", upstream.url.as_str());
         self.add_http_request_header("x-3scale-timeout", &upstream.default_timeout().to_string());
         self.add_http_request_header("x-3scale-service-id", service.id());
-        self.add_http_request_header("x-3scale-service-token", service.token());
+        if service.token.is_none() {
+            // without service token, other info is useless.
+            anyhow::bail!("service token unavailable");
+        }
+        self.add_http_request_header("x-3scale-service-token", service.token().unwrap());
```

Comment:
Same thing here, bail in the else clause of `let token = if let Some(token) = service.token() { ... } else { bail!(...) };`.

Reply:
Done